packages/@expo/server/src/ImmutableRequest.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright © 2025 650 Industries.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * An immutable version of the Fetch API's [`Headers`](https://developer.mozilla.org/en-US/docs/Web/API/Headers) object which prevents mutations.
+ */
+class ImmutableHeaders extends Headers {
+  // TODO(@hassankhan): Merge with `ReadonlyHeaders` from `expo-router`
```

Comment:
nit: I think we'll have to make sure to define iterators so we can pass this to `new Headers(...)`. Not important immediately though